### PR TITLE
fix add package when package directory has spaces

### LINKED ISSUE #10685: `dotnet add package` fails when --packages-directory has spaces
I am running the latest release version 2.2. I tried running the following: 

    dotnet add package packageName --package-directory "D:\Cloud\Dropbox\codes\dotnet samples\nugets"

I get the error: 

> error: Unrecognized command or argument 'samples\nugets'

The error is gone when the folder has no spaces.

### PR DESCRIPTION
**Pull request by Jalalx**
This should close #10685

### CODE REVIEW

**Comment by peterhuene** [comment]
Hi @Jalalx.  Thanks very much for putting this contribution together!

Would you be comfortable adding a `dotnet add package` test that ensures a directory with spaces is respected properly?

Also, you can ignore the CI failure for Debian; it's a known infrastructure issue that we're working on resolving.

**Comment by wli3** [comment]
+ @nkolev92 since it is owned by nuget

**Comment by peterhuene** [comment]
Looks like the test failed on Linux.  I'm investigating.

**Comment by peterhuene** [comment]
I'm going to guess that due to the case sensitive file system, we need lowercase the package name in the directory we're checking, but I'll run it on my Arch system to double check.

File: test/dotnet-add-package.Tests/GivenDotnetPackageAdd.cs [hunk 1]
```diff
@@ -72,6 +72,41 @@ public void
...
+            _output.WriteLine($"[STDOUT] {cmd.StdOut}\n[STDERR]{cmd.StdErr}\n");
+
+            cmd.Should().Pass();
+
+            cmd.StdOut.Should()
+               .Contain($"PackageReference for package \'{packageName}\' version \'{packageVersion}\' added to file '{csproj}'.");
+
+            cmd.StdErr.Should().BeEmpty();
+
+            var restoredPackageDirectory = Path.Combine(packageDirectory, packageName, packageVersion);
```
Comment (peterhuene) [inline test/dotnet-add-package.Tests/GivenDotnetPackageAdd.cs]:
```suggestion
            var restoredPackageDirectory = Path.Combine(packageDirectory, packageName.ToLowerInvariant(), packageVersion);
```

**Comment by peterhuene** [comment]
CI is green (excepting the known issue for the Debian leg).

I'll wait until @nkolev92 has a chance to review/comment before merging.

Thanks, @Jalalx!

**Comment by peterhuene** [comment]
Erm, my eyes are playing tricks on me. The Debian leg was green, it was a seemingly transient failure on the Ubuntu leg.  Re-running.

File: src/dotnet/commands/dotnet-add/dotnet-add-package/Program.cs [hunk 2]
```diff
@@ -143,7 +143,7 @@ private string[] TransformArgs(string packageId, string tempDgFilePath, string p
 
             args.AddRange(_appliedCommand
                 .OptionValuesToBeForwarded()
-                .SelectMany(a => a.Split(' ')));
+                .SelectMany(a => a.Split(' ', 2)));
```
Comment (nkolev92) [inline src/dotnet/commands/dotnet-add/dotnet-add-package/Program.cs]: I don't know the code pretty well, but what happens if I run:

`dotnet add package packageName --package-directory "D:\Cloud\Dropbox\codes\dotnet samples\nugets" --interactive`

or

`dotnet add package packageName --package-directory "D:\Cloud\Dropbox\codes\dotnet samples\nugets" --no-restore`

Comment (peterhuene) [reply src/dotnet/commands/dotnet-add/dotnet-add-package/Program.cs]: That should be fine.  `OptionValuesToBeForwarded` returns an `IEnumerable<string>` for each option, so `SelectMany` should get values like `--package-directory c:\foo bar`, `--interactive`, etc.

Thus, the resulting `argv` will be [`--packge-directory`, `c:\foo bar`, `--interactive`].  Options with spaces in them, like `c:\foo bar` is then quoted depending on the platform rules by other CLI code.

Comment (nkolev92) [reply src/dotnet/commands/dotnet-add/dotnet-add-package/Program.cs]: Ok that makes sense. 
And there are no options that take more than 1 value as far as I can tell.

This seems fine in that case.

**Comment by peterhuene** [comment]
@nkolev92 any other feedback before I click the shiny green button? 😄

**Comment by nkolev92** [comment]
Nope.

**Comment by peterhuene** [comment]
/cdn.vox-cdn.com/uploads/chorus_asset/file/15979212/Clippy_ShipIt.gif)

**Comment by peterhuene** [comment]
@Jalalx Thanks again for your contribution!  It should be available in the next preview release of the 3.0 .NET Core SDK.

**Pull request merged**